Added pagination for GetRoomMessagesReactor

jayesh00949 opened the pull request:
All the messages of the room is getting sent together from the reactor to the frontend which is causing delay in loading the messages to the screen. So, we have added pagination for the response.
Modifications - 
Added two parameters limit and offset in GetRoomMessagesReactor which are optional parameter and will be set to -1 if not sent from frontend.
Overloaded the method "doRetreiveConversation" adding the parameter limit and offset and extracted the common code in addColumns method. In the overloaded method we are setting limit and offset to the query and also sorting it in desc order to get the latest message first.

jayesh00949 committed:
Added pagination for GetRoomMessagesReactor

kunal0137 committed:
Merge branch 'dev' into GetRoomMessagesPagination

kunal0137 committed:
adding additional empty check and renaming addCol method

kunal0137 commented:
cleaned up some minor things but are we sure this is how we want this to work?  im not sure how it would work in actual since you are switching to DESC sorted and then reversing the collection. seems odd but lets walk through an example of how an app would actually use it

jayesh00949 committed:
Added top to down pagination

jayesh00949 committed:
Added top to down pagination

jayesh00949 committed:
merging from remote to local
Merge branch 'GetRoomMessagesPagination' of https://github.com/SEMOSS/Semoss into GetRoomMessagesPagination

jayesh00949 committed:
removed unwanted changes

jayesh00949 committed:
removed unwanted changes from social.properties

jayesh00949 committed:
Added String sort for order flag and removed boolean flag

kunal0137 committed:
Merge branch 'dev' into GetRoomMessagesPagination
